module_utils/network_lsr/argument_validator.py

```diff
@@ -1806,3 +1810,57 @@ def validate_connection_one(self, mode, connections, idx):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+        # Setting ip.dns is not allowed when corresponding IP method for that
+        # nameserver is disabled
+        for nameserver in connection["ip"]["dns"]:
```

Comment:
It would be great to have unit tests to cover the scenarios here.

Reply:
Added unit tests function `test_dns_argvalidator()`